Assignment.Tests/PingProcessTests.cs

```diff
@@ -0,0 +1,260 @@
+using IntelliTect.TestTools;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Assignment.Tests;
+
+[TestClass]
+public class PingProcessTests
+{
+    PingProcess Sut { get; set; } = new();
+    bool IsUnix { get; set; }
+    //Will never be null as its set in TestInitalize
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    string LimitPingArg { get; set; }
+    string PingOutputLikeExpression {get; set;}
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        IsUnix = Environment.OSVersion.Platform is PlatformID.Unix;
+        (string arg, string exp)  = IsUnix ? ("-c", PingOutputLikeExpressionUnix) : ("-n", PingOutputLikeExpressionWindows);
+        LimitPingArg = arg;
+        PingOutputLikeExpression = exp;
+        Sut = new();
+    }
+
+    [TestMethod]
+    public void Start_PingProcess_Success()
+    {
+        Process process = Process.Start("ping", $"{LimitPingArg} 4 localhost");
+        process.WaitForExit();
+        Assert.AreEqual<int>(0, process.ExitCode);
+    }
+
+    [TestMethod]
+    public void Run_GoogleDotCom_Success()
+    {
+
+        int expectedExitCode = Environment.GetEnvironmentVariable("GITHUB_ACTIONS") is not null || IsUnix ? 1 : 0;
+
+        int exitCode = Sut.Run("google.com").ExitCode;
+        Assert.AreEqual<int>(expectedExitCode, exitCode);
+    }
+    
+
+    [TestMethod]
+    public void Run_LocalHost_Success()
+    {
+        int exitCode = Sut.Run("localhost").ExitCode;
+        Assert.AreEqual<int>(0, exitCode);
+    }
+    //When running this test, stdOutput comes back as Empty
+    [TestMethod]
+    public void Run_InvalidAddressOutput_Success()
+    {
+
+        (string expectedOutput, int expectedExitCode)  = IsUnix ? ("ping: badaddress: Temporary failure in name resolution", 2) : ("Ping request could not find host badaddress. Please check the name and try again.",1);
+        (int exitCode, string? stdOutput, string? stdError) = Sut.Run("badaddress");
+        string actualOutput = IsUnix ? stdError! : stdOutput!;
+        //In Unix, error is logged to StdError, not StdOutput
+        Assert.IsFalse(string.IsNullOrWhiteSpace(actualOutput));
+        actualOutput = WildcardPattern.NormalizeLineEndings(actualOutput!.Trim());
+        Assert.AreEqual<string?>(expectedOutput, actualOutput, $"Output is unexpected: {stdOutput}");
+        // 2 is the exit code for invalid address in Unix
+        // 1 is the exit code for invalid address in Windows
+        Assert.AreEqual<int>(expectedExitCode, exitCode);
```

Comment:
```suggestion
        Assert.NotEqual<int>(0, exitCode);
```
This seems easier if we know 0 is always success? but just an idea to reduce complexity